### PR TITLE
Fix filtering by url

### PR DESCRIPTION
**Pull request by mymro**
I am not sure, what the old code tried to do, but it did not work. I changed it to what I believe is the correct implementation.

### CODE REVIEW

File: lncrawl/core/crawler.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def index_of_chapter(self, url: str) -> int:
         """Return the index of chapter by given url or 0"""
         url = self.absolute_url(url)
         for chapter in self.chapters:
-            if chapter.url.rsplit("/") == url:
```
Comment (idMysteries) [inline lncrawl/core/crawler.py:109]: maybe rstrip? @dipu-bd

Comment (mymro) [reply lncrawl/core/crawler.py:109]: Might be it

File: lncrawl/core/crawler.py [hunk 2]
```diff
@@ -106,7 +106,7 @@ def index_of_chapter(self, url: str) -> int:
         """Return the index of chapter by given url or 0"""
         url = self.absolute_url(url)
         for chapter in self.chapters:
-            if chapter.url == url:
+            if chapter.url.rstrip("/") == url.rstrip("/"):
```
Comment (idMysteries) [inline lncrawl/core/crawler.py:109]: ```if chapter.url.rstrip("/") == url:```

bcuz ```url = self.absolute_url(url)```

absolute_url returns url without right "/"

Comment (mymro) [reply lncrawl/core/crawler.py]: Changed it

**Comment by dipu-bd** [comment]
Thanks for finding it out

**Pull request merged**